src/shadowbox/server/server_access_key.ts

```diff
@@ -212,8 +220,8 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
     for (const accessKey of this.accessKeys) {
       const usageBytes = bytesTransferredById[accessKey.id] || 0;
       const wasOverDataLimit = accessKey.isOverDataLimit;
-      accessKey.isOverDataLimit =
-          this.accessKeyDataLimit ? usageBytes > this.accessKeyDataLimit.bytes : false;
+      const limitBytes = (accessKey.dataLimit || this.defaultDataLimit)?.bytes;
+      accessKey.isOverDataLimit = usageBytes > limitBytes || false;
```

Comment:
Good catch on the 0 data limit bug. I don't think you need the `|| false`, since `number > undefined || undefined > undefined == false`.